inference-engine/tests/functional/shared_test_classes/src/read_ir/read_ir.cpp

```diff
@@ -25,101 +19,30 @@ std::string ReadIRTest::getTestCaseName(const testing::TestParamInfo<std::tuple<
-                                                 PSROIPoolNode->get_spatial_bins_y(),
-                                                 PSROIPoolNode->get_mode());
-        } else if (ngraph::is_type<ngraph::op::v0::ROIPooling>(node)) {
-            const auto& ROIPoolNode = std::dynamic_pointer_cast<ngraph::op::v0::ROIPooling>(node);
-            CommonTestUtils::fill_data_roi(blob->buffer(),
-                                           blob->size(),
-                                           ROIPoolNode->get_input_shape(0).front() - 1,
-                                           inputShape[2],
-                                           inputShape[3],
-                                           1.0f,
-                                           ROIPoolNode->get_method() == "max");
-        } else if (ngraph::is_type<ngraph::op::v3::ROIAlign>(node)) {
-            const auto &ROIAlignNode = std::dynamic_pointer_cast<ngraph::op::v3::ROIAlign>(node);
-            std::vector<float> blobData(ROIAlignNode->get_shape()[0] * 4);
-            ROIAlignLayerTest::fillCoordTensor(blobData,
-                                               inputShape[2],
-                                               inputShape[3],
-                                               ROIAlignNode->get_spatial_scale(),
-                                               ROIAlignNode->get_sampling_ratio(),
-                                               ROIAlignNode->get_pooled_h(),
-                                               ROIAlignNode->get_pooled_w());
-            blob = FuncTestUtils::createAndFillBlobWithFloatArray(info.getTensorDesc(), blobData.data(), blobData.size());
-        }
-    } else {
-        blob = GenerateInput(info);
-    }
-    return blob;
 }
 
-void ReadIRTest::Infer() {
-    inferRequest = executableNetwork.CreateInferRequest();
-    inputs.clear();
+void ReadIRTest::GenerateInputs() {
+    auto inputMap = getInputMap();
 
     const auto& inputsInfo = executableNetwork.GetInputsInfo();
     for (const auto& param : function->get_parameters()) {
         const auto infoIt = inputsInfo.find(param->get_friendly_name());
         GTEST_ASSERT_NE(infoIt, inputsInfo.cend());
-
         const auto& info = infoIt->second;
-
-        InferenceEngine::Blob::Ptr blob(nullptr);
         for (size_t i = 0; i < param->get_output_size(); i++) {
             for (const auto& node : param->get_output_target_inputs(i)) {
                 const auto nodePtr = node.get_node()->shared_from_this();
-                if (isRoiOperation(nodePtr)) {
-                    blob = generateROIblob(*info, nodePtr);
-                    break;
+                auto it = inputMap.find(nodePtr->get_type_info());
+                if (it != inputMap.end()) {
+                    inputs.push_back(it->second(nodePtr, *info));
+                } else {
+                    inputs.push_back(GenerateInput(*info));
```

Comment:
Can be even a little bit simpler - by adding `generate(const std::shared_ptr<ngraph::op::v0::PSROIPooling> node,  const InferenceEngine::InputInfo& info)` but the change can be postponed, not a blocker

Reply:
Do you mean `generate(const std::shared_ptr<ngraph::Node> node, const InferenceEngine::InputInfo& info)`? If yes, i have tried to do it, but this one wasn't contained to the map

Reply:
Yep, sorry for a copy-paste issue, I meant  `std::shared_ptr<ngraph::Node>`. Map itself shouldn't contain it, this instance of a `generate` function will be called from `generateInput` if no suitable `generate` method for certain OP type will be found.

Reply:
Agree, polymorphism is in action. Fixed